src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -477,13 +477,192 @@ public sealed class SocketsHttpHandler_HttpClientHandler_Proxy_Test : HttpClient
         protected override bool UseSocketsHttpHandler => true;
     }
 
-    public sealed class SocketsHttpHandler_HttpClientHandler_TrailingHeaders_Test : HttpClientHandlerTest_TrailingHeaders_Test
+    public class SocketsHttpHandler_HttpClientHandler_TrailingHeaders_Test : HttpClientHandlerTest_TrailingHeaders_Test
     {
         public SocketsHttpHandler_HttpClientHandler_TrailingHeaders_Test(ITestOutputHelper output) : base(output) { }
-        // PlatformHandlers don't support trailers.
         protected override bool UseSocketsHttpHandler => true;
     }
 
+    public sealed class SocketsHttpHandler_Http2_HttpClientHandler_TrailingHeaders_Test : HttpClientHandlerTestBase
```

Comment:
Why did you move Http2 trailing header tests here? And not http1? They should be pretty similar.

Reply:
Because these were specific to SocketsHttpHandler and specific to HTTP/2.  Leaving them on the base class then had them be run and then skipped for all other handlers / configurations.  Seemed they should be where they actually target.

Reply:
I still don't understand why protocol version would matter. HttpClientHandlerTest_TrailingHeaders_Test  is only used by SocketsHttpHandlerTest

Reply:
We don't support trailing headers on any other handler?  Then yeah, the HTTP/1.1 variants should be moved to this file as well.

Reply:
I don't think so since the property did not exist before. We made @caesar-chen to put tests for trailing headers to new file.

Reply:
> to put tests for trailing headers to new file.

A separate file is fine.  But these tests are specific to SocketsHttpHandler.  They're not for HttpClientHandler.